### PR TITLE
Improve testkit error readability

### PR DESCRIPTION
**Pull request by andrefmrocha**
Removed logging of cancellation expections coming cancelling the server after tests were finished

Improved readability of testkit assertion related errors, by removing the `ExecutionException` layer and only throwing the assertion error, which improves readability of errors from
```
java.util.concurrent.ExecutionException: Boxed Error
ch.epfl.scala.bsp.testkit.client.TestFailedException: java.util.concurrent.ExecutionException: Boxed Error
(...)
```

to 

```
java.lang.AssertionError: assertion failed: Cleaned cache successfully, when it should have failed
ch.epfl.scala.bsp.testkit.client.TestFailedException: java.lang.AssertionError: assertion failed: Cleaned cache successfully, when it should have failed
(...)
```

### CODE REVIEW

**Review by jastice — APPROVED** [review]

File: testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/mock/MockSession.scala [hunk 1]
```diff
@@ -28,7 +29,7 @@ case class MockSession(
       .create()
 
     val listening = launcher.startListening
-    new Thread (() => listening.get()).start()
+    new Thread (() => Try(listening.get())).start()
```
Comment (jvican) [inline testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/mock/MockSession.scala:32]: Aren't we ignoring any exception thrown here? Why is this necessary?

**Review by jvican — APPROVED** [review]
Thanks for the contribution! LGTM overall 👍 I leave a question though

File: testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/mock/MockSession.scala (hunk 1 above)
Comment (andrefmrocha) [reply testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/mock/MockSession.scala:32]: Any exception thrown here will automatically impact the result of a test, therefore I don't think handling them is necessary. It is, however, wrapped in a `Try` to prevent the output of the `CancellationException` whenever the connection is shutdown, since this exception is irrevelant for the user

Comment (jvican) [reply testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/mock/MockSession.scala:32]: Then the ignoring should be explicit. Let’s replace it with a try catch expression, it’s clearer

Comment (andrefmrocha) [reply testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/mock/MockSession.scala]: Makes sense, changed!

**Review by jvican — APPROVED** [review]
